sdk/core/core/README.md

```diff
@@ -8,182 +8,183 @@ The library allows client libraries to expose common functionality in a consiste
+
+   ```C
+   void test_log_func(az_log_classification classification, az_span message)
+   {
+      printf("%.*s\n", az_span_length(message), az_span_ptr(message));
+   }
+
+   int main()
+   {
+      az_log_classification const classifications[] = { AZ_LOG_HTTP_REQUEST, AZ_LOG_HTTP_RESPONSE, AZ_LOG_END_OF_LIST };
+      az_log_set_classifications(classifications);
+      az_log_set_callback(test_log_func);
 
-## Contributing
-For details on contributing to this repository, see the [contributing guide][azure_sdk_for_c_contributing].
+      // More code goes here...
+   }
+   ```
 
-This project welcomes contributions and suggestions.  Most contributions require you to agree to a
-Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us
-the rights to use your contribution. For details, visit https://cla.microsoft.com.
+### SDK Function Argument Validation
 
-When you submit a pull request, a CLA-bot will automatically determine whether you need to provide
-a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions
-provided by the bot. You will only need to do this once across all repos using our CLA.
+Public SDK functions validate the arguments passed to them in an effort to ensure that calling code is 
+passing valid values. The valid value is called a contract precondition. If an SDK function detects a
+precondition failure (invalid argument value), then by default, it calls a function that places the 
+calling thread into an infinite sleep state; other threads continue to run.
 
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
-For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
-contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+To override the default behavior, implement a function matching the az_precondition_failed_fn function
+signature and then, in your application's initialization (before calling any Azure SDK function), call
+az_precondition_failed_set_callback passing it the address of your function. Now, when any Azure SDK
+function detects a precondition failure, it will invoke your callback instead. You might override the 
+callback to attach a debugger or perhaps to reboot the device rather than allowing it to continue running 
+with unpredictable behavior.
 
-### Additional Helpful Links for Contributors  
-Many people all over the world have helped make this project better.  You'll want to check out:
+Also, if you define the NO_PRECONDITION_CHECKING symbol when compiling the SDK code, all of the Azure SDK 
+precondition checking will be excluding making the binary code smaller and faster. We recommend doing 
```

Comment:
I noticed some things that could be re-worded a bit, when reading the newly added sections. @JeffreyRichter - do you want me to go ahead and update/fix those OR just point them out here and let you fix them? I want to avoid unnecessary merge conflicts in case you have some other content in-flight already.

For example:
```suggestion
precondition checking will be excluded, making the binary code smaller and faster. We recommend doing 
```

Reply:
Yes, please fix them